p2p/host/relay/autorelay_test.go

```diff
@@ -16,6 +16,7 @@ import (
 	autonatpb "github.com/libp2p/go-libp2p-autonat/pb"
 	circuit "github.com/libp2p/go-libp2p-circuit"
 	host "github.com/libp2p/go-libp2p-host"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
```

Comment:
no, we can't depend on kad-dht; this creates a circular reference.

Reply:
@vyzo I brought the `autorelay_test` file into the `relay` package so that I could use the mock functions in the usage example

Reply:
can't do that either; the reason it is not there is that it causes  a circular dependency between the top level package and the test...

Reply:
I see... relay -> libp2p -> config -> relay

I put the examples in `relay_test`. This seems a little brittle, no?

Reply:
it's ok, it's more of a go failing.